### PR TITLE
fix: issue with color contrast for tag

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

DBTag (as checkbox) wasn't accessible in force-colors mode

## Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

**Comment by mfranzke** [comment]
@nmerget this needs a snapshot update.

File: packages/components/src/components/tag/tag.scss [hunk 1]
```diff
@@ -92,12 +92,20 @@
 		display: flex;
 		align-items: center;
 
+		@media (forced-colors: active) {
+			border-style: dashed;
```
Comment (mfranzke) [inline packages/components/src/components/tag/tag.scss:96]:
```suggestion
			&:not(:has(input:checked))
				border-style: dashed;
			}
```

File: packages/components/src/components/tag/tag.scss [hunk 2]
```diff
@@ -92,12 +92,20 @@
...
 		&:has(input:checked) {
 			border-width: variables.$db-border-height-3xs;
 
 			&::before {
 				content: none;
 			}
+
+			@media (forced-colors: active) {
+				border-style: solid;
+			}
```
Comment (mfranzke) [inline packages/components/src/components/tag/tag.scss:108]:
```suggestion
```

In combination with the previous suggestion.

**Comment by mfranzke** [comment]
@leape is this aligned with you to have a [dashed border for unchecked state vs. checked state](https://github.com/db-ui/mono/blob/305f65b9b96b1b9c63406c44af9a4efed143d2f6/__snapshots__/tag/showcase/chromium-highContrast/DBTag-should-match-screenshot-1/DBTag-should-match-screenshot.png) on High Contrast Mode?

File: packages/components/src/components/tag/tag.scss (hunk 1 above)
Comment (nmerget) [reply packages/components/src/components/tag/tag.scss]: fixed

File: packages/components/src/components/tag/tag.scss (hunk 2 above)
Comment (nmerget) [reply packages/components/src/components/tag/tag.scss]: fixed

**Review by mfranzke — APPROVED** [review]

**Pull request merged**